display_list/display_list_image_filter.h

```diff
@@ -137,6 +153,26 @@ class DlImageFilter
                                            const SkMatrix& ctm,
                                            SkIRect& input_bounds) const = 0;
 
+  MatrixCapability get_matrix_capability() const {
+    MatrixCapability result = this->matrix_capability();
+    // CropRects need to apply in the source coordinate system, but are not
+    // aware of complex CTMs when performing clipping. For a simple fix, any
+    // filter with a crop rect set cannot support more than scale+translate CTMs
+    // until that's updated. if (this->cropRectIsSet()) {
+    //     result = std::min(result, MatrixCapability::kScaleTranslate);
+    // }
+    for (auto* filter : filters()) {
+      result = std::min(result, filter->matrix_capability());
+    }
+    return result;
+  }
+
+  virtual MatrixCapability matrix_capability() const {
+    return MatrixCapability::kScaleTranslate;
+  }
+
+  virtual std::vector<const DlImageFilter*> filters() const { return {this}; }
```

Comment:
This is actually unnecessary for child-less filters - their own method was already called before this vector is considered so this default implementation just forces their own method to be called twice. An empty vector would be less redundant.

But, really, I think this vector solution is overkill for our current minimal tree.

Reply:
Why is this method still here? It is unused. Please delete it.